### ISSUE TITLE
[Question] Performance & RAM usage

### ISSUE THREAD

**Issue by ReimuNotMoe** [markqvist/Reticulum#437]
Hi, I would like to get Reticulum running on a MT7628 router board with 64MB RAM. Do you think the CPU performance & RAM size are enough?

**Comment by faragher** [comment]
No. The current Python implementation, at idle, is around 400 megs according to top. I had a 1 gig server that ran well until a file transfer was requested, then it would crash due to lack of memory. Upgrading to 4 gigs shows it peaks just under 2 gigs. So for a simple router, I'd say 1 gig RAM is about the minimum, maybe 512 with a super lean OS. 2-4 gigs is the best bet for something running a node.

Other implementations aren't complete, and we can't judge their memory footprint at this time. I expect them to be lower once mature, but until they're complete, it's not very helpful to speculate.
